src/components/logo/logo.js

```diff
@@ -0,0 +1,77 @@
+//@flow
+import style from '../../styles/style.scss';
+import {h} from 'preact';
+import {connect} from 'preact-redux';
+import BaseComponent from '../base';
+import {Text} from 'preact-i18n';
+import {PLAYER_SIZE} from '../shell/shell';
+
+/**
+ * mapping state to props
+ * @param {*} state - redux store state
+ * @returns {Object} - mapped state to this component
+ */
+const mapStateToProps = state => ({
+  isMobile: state.shell.isMobile,
+  playerSize: state.shell.playerSize,
+  config: state.config.components.logo
+});
+
+@connect(mapStateToProps)
+/**
+ * LogoControl component
+ *
+ * @class LogoControl
+ * @example <LogoControl player={this.player} />
+ * @extends {BaseComponent}
+ */
+class LogoControl extends BaseComponent {
+  /**
+   * @static
+   * @type {string} - Component display name
+   */
+  static displayName = 'logo';
+  /**
+   * should render component
+   * @param {*} props - component props
+   * @returns {boolean} - whether to render the component
+   * @static
+   */
+  static shouldRender(props: any): boolean {
+    const componentConfig = props.config.components[this.displayName];
+    return !(Object.keys(componentConfig).length === 0 && componentConfig.constructor === Object);
```

Comment:
what is this check?

Reply:
this is for case that we didn't config this ui element so we didn't want to render the element.
I took it actually from watermark and I saw it in other configurable components

Reply:
yes, I saw the reference but couldn't make sense of the actual check itself. What does it check for and why like this? you can check with @yairans who created it for the other component.